gammapy/cube/models.py

```diff
@@ -277,12 +278,32 @@ def compute_flux(self):
         flux = dnde * volume
         return flux.to('cm-2 s-1')
 
+    def apply_aeff(self, flux):
+        """Compute npred cube
+
+        For now just divide flux cube by exposure
+        """
+        data = flux * self.exposure.quantity
+        flux = Map.from_geom(self.geom, unit=data.unit)
```

Comment:
There should be a nicer way to create a Map from a ``geom`` and a ``data`` object (without caring if it's wcs of healpix or god knows what ...). Did I just not find it, or doesn't it exist?

Reply:
Ideally we only have Numpy array evaluations, and no Map creation in our likelihood loop, no?
Remove this method again?

Making new Map objects is just something that could sometimes be done for debugging, bug shouldn't happen all the time in the likleihood evaluation, no?

Yes, we need to discuss map copy and creation a bit more and probably add one or two more methods to make that convenient; just put something that works for now.

Reply:
OK. Note that the `PSFKernel.apply()` method produces a new `Map`. This might be changed if needed.

Reply:
I think the kernel numpy array should be exposed as a property on the `PSFKernel`, and then directly passed to the scipy or astropy convolve call. IMO a goal is to only have Numpy in the model and likelihood evaluation, no more Astropy or Gammapy objects. That means it'll be fast and easy to translate to Tensorflow in a few years when the time has come. :-)

Reply:
The problem here is that ``PSFKernel.apply`` takes [only a map](https://github.com/gammapy/gammapy/blob/master/gammapy/cube/psf_kernel.py#L255) ... Can this easily be changed @registerrier ?

Reply:
How about, for now, do an in-place apply, i.e. don't make a new map in the method?
(there's no callers for this yet, you can make any changes you like)

Reply:
Passing a map for convolution has some virtues since you can directly check the consistency of the two geometries used and take advantage of the image iterator of `Map`. 
Maybe it would be better to have a function/method living directly on `Map` to perform map convolutions rather than having this in the PSFKernel.

Anyway, I can make an in-place convolution and also expose the PSFKernel array through a property. If this helps. In that case I suppose that convolution could also be implemented in the `SkyModelMapEvaluator`.

Reply:
For now, suggest to leave as-is. (make the apply in-place if you like).

> Maybe it would be better to have a function/method living directly on Map to perform map convolutions rather than having this in the PSFKernel.

This might be better, especially if we do in-place apply.
I think we're already coupling to `EvenList` in the fill method.
But it requires some thought how `gammapy.maps` couples to `gammapy.irf` or `gammapy.cube` to avoid circular dependencies or complexity.
Keeping gammapy.maps simple and basic, and only coupling with other parts of Gamampy in higher-up modules also has advantages.
Let's discuss this week ...